### PR TITLE
App Args: Add tests, fix parsing of LitInt in application arguments

### PR DESCRIPTION
**Pull request by richardeoin**
It appears that `input.parse::<LitBool>()` consumes part of the token
stream, so the subsequent call to `input.parse::<LitInt>()` does not
behave as expected.

This PR has one possible fix, but there may be a better fix using
`fork` or `step` (see https://docs.rs/syn/1.0.11/syn/parse/struct.ParseBuffer.html#method.fork)

Add tests.

### CODE REVIEW

File: src/parse/app.rs [hunk 1]
```diff
@@ -70,27 +70,30 @@ impl AppArgs {
...
-                                ident.span(),
-                                "argument has unexpected value",
-                            ));
+                            // Parse as literal
+                            match input.parse::<Lit>()? {
+                                Lit::Bool(lit) => {
+                                    custom.insert(ident, CustomArg::Bool(lit.value));
+                                }
+                                Lit::Int(lit) => {
+                                    if lit.suffix().is_empty() {
```
Comment (japaric) [inline src/parse/app.rs:83]: if this comparison fails we should return the "argument has unexpected value" error as before, or even better we should return an "integer must be unsuffixed" error

**Review by japaric — CHANGES_REQUESTED** [review]
Thanks for the PR! There's one small issue that needs to be addressed and then we can merge this.

File: src/parse/app.rs (hunk 1 above)
Comment (richardeoin) [reply src/parse/app.rs:83]: Cheers for the review! Squashed and pushed below

**Comment by japaric** [comment]
Excellent! Thank you.

bors r+

**Comment by richardeoin** [comment]
> Excellent! Thank you.
> 
> bors r+

Great 👍

Bors appears to have failed. What happened?

**Review by japaric — APPROVED** [review]
bors r+

**Comment by japaric** [comment]
@richardeoin ah, sorry. I forgot to change the state of the review from "needs changes" to "approved" before sending this PR to bors that's why it didn't get processed by bors.

**Pull request merged**